src/main.rs

```diff
@@ -72,33 +80,44 @@ async fn main() -> Result<(), APIError> {
         _ => loader.download_model(model_id, None, args.hf_token, args.hf_token_path)?,
     };
 
-    let dtype = match args.dtype.as_deref() {
-        Some("f16") => DType::F16,
-        Some("bf16") => DType::BF16,
-        Some("f32") => DType::F32,
+    let (dtype, dsize) = match args.dtype.as_deref() {
+        Some("f16") => (DType::F16, 2),
+        Some("bf16") => (DType::BF16, 2),
+        Some("f32") => (DType::F32, 4),
         Some(dtype) => panic!("Unsupported dtype {dtype}"),
-        None => DType::BF16,
+        None => (DType::BF16, 2),
     };
 
     let device = candle_examples::device(args.cpu).unwrap();
     let model = loader.load_model(paths, dtype, device)?;
+    let config = model.0.get_model_config();
+    let num_gpu_blocks = args.kvcache_mem * 1024 * 1024
```

Comment:
Can we change the 1024 * 1024 to a constant with a name that describes the conversion (MB -> B)?

Reply:
> Can we change the 1024 * 1024 to a constant with a name that describes the conversion (MB -> B)?

No problem.